subsys/bluetooth/controller/ll_sw/nordic/lll/lll_clock.c

```diff
@@ -66,10 +66,27 @@ int lll_clock_init(void)
 
 int lll_clock_wait(void)
 {
-	struct onoff_manager *mgr =
-		z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_LF);
+	struct onoff_manager *mgr;
+	static bool done;
+	int err;
+
+	if (done) {
+		return 0;
+	}
+	done = true;
 
-	return blocking_on(mgr, LFCLOCK_TIMEOUT_MS);
+	mgr = z_nrf_clock_control_get_onoff(CLOCK_CONTROL_NRF_SUBSYS_LF);
+	err = blocking_on(mgr, LFCLOCK_TIMEOUT_MS);
+	if (err) {
+		return err;
+	}
+
+	err = onoff_release(mgr);
+	if (err <= 0) {
+		return -EIO;
+	}
```

Comment:
@nordic-krch updated to fail if 0 or less, as it is required that LF clock remains on.